lib/interfaces/plugins.js

```diff
@@ -14,153 +14,191 @@
  * limitations under the License.
 */
 
-var debug = require('debug')('signalk:interfaces:plugins')
-var fs = require('fs')
-var path = require('path')
-var express = require('express')
+var debug = require("debug")("signalk:interfaces:plugins");
+var fs = require("fs");
+var path = require("path");
+var express = require("express");
 
 module.exports = function(app) {
-
   return {
     start: function() {
-      startPlugins(app)
+      startPlugins(app);
 
       ensureExists(path.join(__dirname, "../../plugin-config-data"));
 
-      app.use('/plugins/configure', express.static(path.join(__dirname, '/../../plugin-config/public')));
+      app.use(
+        "/plugins/configure",
+        express.static(path.join(__dirname, "/../../plugin-config/public"))
+      );
 
       router = express.Router();
 
-      app.get('/plugins', function(req, res, next) {
-        res.json(app.plugins.map((plugin) => {
-          var data = null
-          try {
-            data = getPluginOptions(plugin.id)
-          } catch(e) {
-            console.log(e.code + " " + e.path)
-          }
-          return {
-            id: plugin.id,
-            name: plugin.name,
-            version: plugin.version,
-            description: plugin.description,
-            schema: plugin.schema,
-            uiSchema: plugin.uiSchema,
-            state: plugin.state,
-            data: data
-          }
-        }))
+      app.get("/plugins", function(req, res, next) {
+        res.json(
+          app.plugins.map(plugin => {
+            var data = null;
+            try {
+              data = getPluginOptions(plugin.id);
+            } catch (e) {
+              console.log(e.code + " " + e.path);
+            }
+            var uiSchema = plugin.uiSchema;
+            if (
+              typeof uiSchema === "undefined" ||
+              typeof uiSchema["configuration"] === "undefined" ||
+              typeof uiSchema["configuration"]['ui:order'] === "undefined"
+            ) {
+              var order = [];
+              Object.keys(plugin.schema.properties).forEach(name => {
```

Comment:
Either make this work properly for nested properties or scrap it - I don't think it makes sense to have it work for only top level properties.

And the reasoning for this ("json order is not guaranteed and on ios is not preserved in practice") should be included in the commit description or in PR.